src/shogun/structure/MAPInference.cpp

```diff
@@ -38,6 +39,9 @@ CMAPInference::CMAPInference(CFactorGraph* fg, EMAPInferType inference_method)
 		case GRAPH_CUT:
```

Comment:
Now I think it's safe to remove other inference types, only keep TreeMaxProd (for tree), GraphCut (for binary loopy graph, better with submodular energy gurantee) and GEMPLP (general loopy graph), since we don't want to implement more methods. The default one choose GEMPLP. In a short time period, we don't want to support 3rd party inference lib.  
@Jiaolong @iglesias Thoughts?